src/lua/api-gateway/aws/AwsService.lua

```diff
@@ -187,7 +187,12 @@ function _M:getHttpClient()
 end
 
 function _M:getAWSHost()
-    return self.aws_service .. "." .. self.aws_region .. ".amazonaws.com"
+    -- FIXME: the endpoint cannot always be determined mechanically; http://docs.aws.amazon.com/general/latest/gr/rande.html
```

Comment:
I needed a quick access to S3 service with authentication at that moment and I didn't know what these are supposed to be like, so I decided not to create a "subclass" for it.  Overriding the function perfectly makes sense to me.  That said, I'd bet the part of the code which determines the correct endpoint for a given service would be easier to maintain if it was centralized in one place.